python/plugins/processing/modeler/ModelerParameterDefinitionDialog.py

```diff
@@ -413,8 +460,9 @@ def accept(self):
             self.param.setDescription(name)
             self.param.setMetadata(paramTypeDef.metadata())
 
-        if not self.requiredCheck.isChecked():
-            self.param.setFlags(self.param.flags() | QgsProcessingParameterDefinition.FlagOptional)
+        if not isinstance(self.param, QgsProcessingDestinationParameter):
```

Comment:
Also remove this check to allow non-mandatory outputs in models

Reply:
Same thing, mandatory for the model output is given by the child algorithm output.